src/lib/protocols/tls.c

```diff
@@ -1365,6 +1373,60 @@ int processClientServerHello(struct ndpi_detection_module_struct *ndpi_struct,
 		    }
 		  }
 		}
+	      } else if(extension_id == 65445 /* QUIC transport parameters */) {
+		u_int16_t s_offset = offset+extension_offset;
+		uint16_t final_offset;
+		int using_var_int = is_version_with_var_int_transport_params(quic_version);
+
+		if(!using_var_int) {
+		  if(s_offset+1 >= total_len) {
+		    final_offset = 0; /* Force skipping extension */
+		  } else {
+		    u_int16_t seq_len = ntohs(*((u_int16_t*)&packet->payload[s_offset]));
+		    s_offset += 2;
+	            final_offset = MIN(total_len, s_offset + seq_len);
+		  }
+		} else {
+	          final_offset = MIN(total_len, s_offset + extension_len);
+		}
+
+		while(s_offset < final_offset) {
+		  u_int64_t param_type, param_len;
+
+                  if(!using_var_int) {
+		    if(s_offset+3 >= final_offset)
```

Comment:
Not sure here, but shouldn't it be `s_offset+4 >= final_offset` or `s_offset+3 > final_offset` ?

Reply:
Neither am I sure... I read 4 bytes from s_offset to s_offset + 3 included. The last byte read should be valid, i.e. < final_offset. So the test seems correct... Am I missing something?

Reply:
You are right. Maybe it is just a LGTM false positive.

Reply:
Can I merge this code?

Reply:
There is nothing to argue against merging this PR from my side.

Reply:
> There is nothing to argue against merging this PR from my side.

I agree

Reply:
@IvanNardi Do you think other TAG/Values may be extracted in the future and stored within a quic field in ndpi_flow_struct? For example SCID or ConnID. This can be useful to track flows belonging to the same QUIC session and also QUIC user ID on the server-side.

Reply:
I'm thinking of it in scenarios where an app using nDPI will export QUIC tags as a Json or whatever. What do you think about it?